File: AD-HOC/1553/main.py
def get_frequencies(p):
    frequencies = {}
    for question in p:
        frequencies[question] = frequencies[question] + 1 if question in frequencies else 1
    return frequencies.values()


def get_questions(frequencies, k):
    questions = 0
    for frequency in frequencies:
        questions += frequency >= k
    return questions


n, k = map(int, input().split())
while n > 0 and k > 0:
    p = map(int, input().split())
    frequencies = get_frequencies(p)
    print(get_questions(frequencies, k))
    n, k = map(int, input().split())

File: AD-HOC/1553/main.dart
import 'dart:io';

Iterable<int> getFrequencies(Iterable<int> p) {
  Map<int, int> frequencies = {};
  for (int question in p) {
    frequencies[question] = frequencies.containsKey(question) ? frequencies[question]! + 1 : 1;
  }
  return frequencies.values;
}

int getQuestions(Iterable<int> frequencies, int k) {
  int questions = 0;
  for (int frequency in frequencies) {
    if (frequency >= k) questions++;
  }
  return questions;
}

void main() {
  var [n, k] = stdin.readLineSync()!.split(' ').map(int.parse).toList();
  while (n > 0 && k > 0) {
    Iterable<int> p = stdin.readLineSync()!.split(' ').map(int.parse);
    Iterable<int> frequencies = getFrequencies(p);
    print(getQuestions(frequencies, k));
    [n, k] = stdin.readLineSync()!.split(' ').map(int.parse).toList();
  }
}
